File: frontend-web/src/app/core/services/dashboard.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { BaseResponse } from '../models/auth.model';

export interface DashboardResumoDto {
  totalAlunos: number;
  turmasAtivas: number;
  presencasHoje: number;
  alunosInadimplentes: number;
  proximasAulas: ProximaAulaDto[];
}

export interface FrequenciaDiariaDto {
  data: string;
  total: number;
}

export interface ProximaAulaDto {
  horarioId: string;
  turma: string;
  modalidade: string;
  horaInicio: string;
  horaFim: string;
  sala?: string;
  totalAlunos: number;
  capacidadeMaxima: number;
}

@Injectable({ providedIn: 'root' })
export class DashboardService {
  private readonly http = inject(HttpClient);

  getResumo(): Observable<BaseResponse<DashboardResumoDto>> {
    return this.http.get<BaseResponse<DashboardResumoDto>>(
      `${environment.apiUrl}/api/dashboard/resumo`
    );
  }

  getFrequencia(dias = 14): Observable<BaseResponse<FrequenciaDiariaDto[]>> {
    return this.http.get<BaseResponse<FrequenciaDiariaDto[]>>(
      `${environment.apiUrl}/api/dashboard/frequencia?dias=${dias}`
    );
  }
}

File: backend/src/AcademiaFight.Application/Services/DashboardService.cs
using AcademiaFight.Application.Common;
using AcademiaFight.Application.DTOs.Dashboard;
using AcademiaFight.Application.Helpers;
using AcademiaFight.Application.Interfaces;
using AcademiaFight.Domain.Enums;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Logging;

namespace AcademiaFight.Application.Services;

public class DashboardService : IDashboardService
{
    private readonly IAppDbContext _db;
    private readonly ILogger<DashboardService> _logger;

    public DashboardService(IAppDbContext db, ILogger<DashboardService> logger)
    {
        _db = db;
        _logger = logger;
    }

    public async Task<BaseResponse<DashboardResumoDto>> GetResumoAsync(CancellationToken ct = default)
    {
        var hoje = DateTimeHelper.Hoje();
        var horaAgora = DateTimeHelper.HoraAgora();
        var diaSemanaHoje = (DiaSemana)(int)DateTime.Today.DayOfWeek;

        var totalAlunos = await _db.Usuarios
            .CountAsync(u => u.Perfil == PerfilUsuario.Aluno && u.Ativo, ct);

        var turmasAtivas = await _db.Turmas.CountAsync(t => t.Ativo, ct);
        var presencasHoje = await _db.Presencas.CountAsync(p => p.Data == hoje, ct);

        var alunosSemTurma = await _db.Usuarios
            .Where(u => u.Perfil == PerfilUsuario.Aluno && u.Ativo)
            .CountAsync(u => !_db.Matriculas.Any(m => m.AlunoId == u.Id && m.Ativo), ct);

        var proximasAulas = await _db.Horarios
            .Include(h => h.Turma).ThenInclude(t => t.Modalidade)
            .Include(h => h.Turma).ThenInclude(t => t.Matriculas.Where(m => m.Ativo))
            .Where(h => h.DiaSemana == diaSemanaHoje && h.HoraInicio >= horaAgora)
            .OrderBy(h => h.HoraInicio)
            .Take(3)
            .ToListAsync(ct);

        return BaseResponse<DashboardResumoDto>.Ok(new DashboardResumoDto
        {
            TotalAlunos = totalAlunos,
            TurmasAtivas = turmasAtivas,
            PresencasHoje = presencasHoje,
            AlunosInadimplentes = alunosSemTurma,
            ProximasAulas = proximasAulas.Select(h => new ProximaAulaDto
            {
                HorarioId = h.Id,
                Turma = h.Turma?.Nome ?? string.Empty,
                Modalidade = h.Turma?.Modalidade?.Nome ?? string.Empty,
                HoraInicio = h.HoraInicio,
                HoraFim = h.HoraFim,
                Sala = h.Sala,
                TotalAlunos = h.Turma?.Matriculas?.Count(m => m.Ativo) ?? 0,
                CapacidadeMaxima = h.Turma?.CapacidadeMaxima ?? 0
            })
        });
    }

    public async Task<BaseResponse<IEnumerable<FrequenciaDiariaDto>>> GetFrequenciaDiariaAsync(int dias, CancellationToken ct = default)
    {
        var hoje = DateTimeHelper.Hoje();
        var inicio = hoje.AddDays(-(dias - 1));

        var dados = await _db.Presencas
            .Where(p => p.Data >= inicio && p.Data <= hoje)
            .GroupBy(p => p.Data)
            .Select(g => new FrequenciaDiariaDto { Data = g.Key, Total = g.Count() })
            .ToListAsync(ct);

        // Preenche dias sem presenças com zero
        var resultado = Enumerable.Range(0, dias)
            .Select(i => inicio.AddDays(i))
            .Select(d => dados.FirstOrDefault(x => x.Data == d) ?? new FrequenciaDiariaDto { Data = d, Total = 0 })
            .ToList();

        return BaseResponse<IEnumerable<FrequenciaDiariaDto>>.Ok(resultado);
    }
}
